server/fleet/microsoft_mdm.go

```diff
@@ -820,77 +823,373 @@ func (e MDMWindowsEnrolledDevice) AuthzType() string {
+)
+
+// Supported protocol command verbs
+const (
+	CmdAdd     = "Add"     // Protocol Command verb Add
+	CmdAlert   = "Alert"   // Protocol Command verb Alert
+	CmdAtomic  = "Atomic"  // Protocol Command verb Atomic
+	CmdDelete  = "Delete"  // Protocol Command verb Delete
+	CmdExec    = "Exec"    // Protocol Command verb Exec
+	CmdGet     = "Get"     // Protocol Command verb Get
+	CmdReplace = "Replace" // Protocol Command verb Replace
+	CmdResults = "Results" // Protocol Command verb Results
+	CmdStatus  = "Status"  // Protocol Command verb Status
+)
+
+// ProtoCmdOperation is the abstraction to represent a SyncML Protocol Command
+type ProtoCmdOperation struct {
+	Verb string    `db:"verb"`
+	Cmd  SyncMLCmd `db:"cmd"`
 }
 
-type SyncMLCommand struct {
-	XMLName xml.Name
-	CmdID   int                 `xml:",omitempty"`
-	MsgRef  string              `xml:",omitempty"`
-	CmdRef  string              `xml:",omitempty"`
-	Cmd     string              `xml:",omitempty"`
-	Target  string              `xml:"Target>LocURI"`
-	Source  string              `xml:"Source>LocURI"`
-	Data    string              `xml:",omitempty"`
-	Item    []SyncMLCommandItem `xml:",any"`
+// Protocol Command
+type SyncMLCmd struct {
+	XMLName xml.Name   `xml:",omitempty"`
+	CmdID   string     `xml:"CmdID"`
+	MsgRef  *string    `xml:"MsgRef,omitempty"`
+	CmdRef  *string    `xml:"CmdRef,omitempty"`
+	Cmd     *string    `xml:"Cmd,omitempty"`
+	Data    *string    `xml:"Data,omitempty"`
+	Items   *[]CmdItem `xml:"Item,omitempty"`
 }
 
-type SyncMLBody struct {
-	Item []SyncMLCommand `xml:",any"`
+type CmdItem struct {
+	Source *string `xml:"Source>LocURI,omitempty"`
+	Target *string `xml:"Target>LocURI,omitempty"`
+	Meta   *Meta   `xml:"Meta,omitempty"`
+	Data   *string `xml:"Data"`
 }
 
-// IsValidSyncMLMsg checks for required fields in the SyncML message
-func (req *SyncMLMessage) IsValidSyncMLMsg() error {
-	if req == nil {
-		return errors.New("invalid SyncML message: nil")
+type Meta struct {
+	Type   *MetaAttr `xml:"Type,omitempty"`
+	Format *MetaAttr `xml:"Format,omitempty"`
+}
+
+type MetaAttr struct {
+	XMLNS   string  `xml:"xmlns,attr"`
+	Content *string `xml:",chardata"`
+}
+
+type LocURI struct {
+	LocURI *string `xml:",omitempty"`
+}
+
+func (msg *SyncML) IsValidHeader() error {
+	if strings.TrimSpace(msg.Xmlns) == "" {
+		return errors.New("msg namespace")
+	}
+
+	// SyncML DTD version check
+	if msg.SyncHdr.VerDTD != mdm.SyncMLSupportedVersion {
+		return errors.New("unsupported DTD version")
 	}
 
-	if len(req.Header.Version) == 0 {
-		return errors.New("invalid SyncML message: Version")
+	// SyncML Proto version check
+	if msg.SyncHdr.VerProto != mdm.SyncMLVerProto {
+		return errors.New("unsupported proto version")
 	}
 
-	if len(req.Header.Target) == 0 {
-		return errors.New("invalid SyncML message: Target")
+	// SyncML SessionID check
+	if utf8.RuneCountInString(msg.SyncHdr.SessionID) < 1 {
```

Comment:
Looks like we could just do this?

```suggestion
	if msg.SyncHdr.SessionID == "" {
```